lib/data/models/nostr_event.dart

```diff
@@ -100,13 +99,36 @@ extension NostrEventExtensions on NostrEvent {
       pubkey,
     );
 
-    final rumorEvent = NostrEvent.deserialized(
+    final innerEvent = NostrEvent.deserialized(
       '["EVENT", "", $decryptedContent]',
     );
-    if (rumorEvent.kind != 1) {
-      throw Exception('Not a Mostro DM: ${rumorEvent.toString()}');
+
+    if (innerEvent.kind == 13) {
+      try {
+        final messageContent = await NostrUtils.decryptNIP44(
+          innerEvent.content!,
+          receiver.private,
+          innerEvent.pubkey,
+        );
+
+        final messageEvent = NostrEvent.deserialized(
+          '["EVENT", "", $messageContent]',
+        );
+
+        if (messageEvent.kind != 14) {
+          throw Exception(
+              'Not a NIP-17 direct message: ${messageEvent.toString()}');
+        }
+
+        return messageEvent;
+      } catch (e) {
+        return innerEvent;
+      }
+    } else if (innerEvent.kind == 1) {
+      return innerEvent;
+    } else {
+      return innerEvent;
```

Comment:
Mostro Chat DMs are _not_ NIP-17 messages, they are a variant using a shared key as described in the Mostro Protocol Documentation: https://mostro.network/protocol/chat.html